[Checkpoint] Rename file

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #72754
* __->__ #72748

Removes underscore from file/class as directory is already private

Differential Revision: [D34179308](https://our.internmc.facebook.com/intern/diff/D34179308/)

rohan-varma committed:
[Checkpoint] Rename file

Removes underscore from file/class as directory is already private

Differential Revision: [D34179308](https://our.internmc.facebook.com/intern/diff/D34179308/)

[ghstack-poisoned]

mrshenli commented on torch/distributed/algorithms/_checkpoint/checkpoint_wrapper.py:
```diff
@@ -11,7 +11,7 @@ class CheckpointImpl(Enum):
 
 
-class _CheckpointWrapper(torch.nn.Module):
+class CheckpointWrapper(torch.nn.Module):
```
this class is never used in tests?

mrshenli reviewed (approved):
LGTM! Thanks for prompt fixes.

rohan-varma committed:
Update on "[Checkpoint] Rename file"


Removes underscore from file/class as directory is already private

Differential Revision: [D34179308](https://our.internmc.facebook.com/intern/diff/D34179308/)

[ghstack-poisoned]
